src/hotspot/share/utilities/ostream.cpp

```diff
@@ -589,8 +589,11 @@ char* fileStream::readln(char *data, int count ) {
   char * ret = NULL;
   if (_file != NULL) {
     ret = ::fgets(data, count, _file);
-    //Get rid of annoying \n char
-    data[::strlen(data)-1] = '\0';
+    // Get rid of annoying \n char only if it presents, it works for Posix
+    // and Windows since the last character of these systems is always '\n'
+    if (data[::strlen(data)-1] == '\n') {
+      data[::strlen(data)-1] = '\0';
```

Comment:
Perhaps:
```
    size_t last_char = ::strlen(data) - 1;
    if (last_char >= 0 && data[last_char] == '\n') {
      data[last_char] = '\0';
```

Reply:
@dcubed-ojdk size_t is unsigned so by definition always >= 0. :) I suggest:
`size_t len = ::strlen(data);`
`if (len > 0 && data[len - 1] == '\n') {`
`  data[len - 1] = '\0';`
`}`

Reply:
Make sense. Changed.